metaflow/plugins/kfp/argo_util.py

```diff
@@ -0,0 +1,59 @@
+import shutil
+import subprocess
+import warnings
+
+from kfp.components._yaml_utils import dump_yaml
+
+
+argo_path = shutil.which("argo")
+
+
+def validate_workflow(workflow: dict, kinds: str = "workflows"):
+    if argo_path:
+        has_working_argo_lint = False
+        try:
+            has_working_argo_lint = run_argo_lint(sample_argo_workflow, "workflows")
+        except:
+            warnings.warn(
+                "Cannot validate the compiled workflow."
+                "Found the argo program in PATH, but it's not usable."
+                "Argo CLI v3.1.1+ should work."
+            )
+
+        if has_working_argo_lint:
+            run_argo_lint(dump_yaml(workflow), kinds=kinds)
+
+
+def run_argo_lint(yaml_text: str, kinds: str):
+    # Running Argo lint if available
+    if argo_path:
+        result = subprocess.run(
+            [
+                argo_path,
```

Comment:
This expects and requires `argo` to be installed.  Must we do that for compile?
Could we move the validation to unit tests, because if `argo lint` passes in unit tests, it would continue to pass in CICD.

Also, project images would need this for their `doit compile`

Reply:
If no argo cli is installed this will not be ran. Also if an incompatible argo CLI is found it only emits an warning instead of failing. We may have customers compile Argo WF / WF template from command line. Adding it here checks for the compiled output.

Also it is already the behavior we have with or without `--yaml-only`. It's called by KFP here: https://github.com/zillow/pipelines/blob/feature/zg/sdk/python/kfp/compiler/compiler.py#L1232
Since I'm directly calling create and write workflow, it might make sense that I replicate the same behavior here.